db/view/view.cc

```diff
@@ -2097,7 +2098,9 @@ future<> view_builder::drain() {
 
 future<> view_builder::stop() {
     vlogger.info("Stopping view builder");
-    return drain();
+    // If the view builder has been started, verify that it's also been drained before we stop it.
+    SCYLLA_ASSERT(!_started_flag || _as.abort_requested());
+    co_return;
```

Comment:
> Before these changes, we only drained the view builder when stopping it,

No, we also drain v.b. in storage_service::do_drain(). Why is it not enough?

> which was paired with its *construction*. However, since the view
builder depends on group0_service, which is initialized AFTER the view
builder is constructed, that could lead to segmentation faults.

Then group0 shouldn't be initialized _after_ a service that depends on it. Why can't group0 service be started before view builder?